docs/project/release-notes.md

```diff
@@ -0,0 +1,219 @@
+using `languagePluginURL` and `languagePluginLoader` is also deprecated, use
+instead {any}`globalThis.loadPyodide`. Access to Python globals via
+{any}`pyodide.globals` has also changed: `pyodide.globals.x` ==>
+`pyodide.globals.get("x")` (`pyodide.globals.x` is still supported but is
+deprecated).
+
+### Changes to type translations
+In the past we found that one of the major pain points in using Pyodide occurred
+when an object makes a round trip from Python to Javascript and back to Python
+and comes back different. This violates the expectations of the user and forces
+inelegant workarounds (see {issue}`780` and {issue}`892` among others).
+
+The type conversion module has significantly reworked in v0.17 with the goal
+that round trip conversions of objects between Python and Javascript produces an
+identical object. That is, Python -> JS -> Python conversion produces an object
+that's now equal to the original object, and JS -> Python -> JS conversion
+verifies the `===` equality operation.
+
+We also made extensive additions to the type conversions test suite and
+documentation, though gaps remain.
+
+See issue {issue}`900` for some of the discussion.
+
+(Mostly implemented in PRs {pr}`1152` and {pr}`1167`, see also {pr}`1186` which )
+
+### Changes to buffer translations
+
+The old buffer translation code was inflexible, leaky, and had serious bugs
+including use after free ({issue}`749`) and buffer overflow errors.
+
+We completely reworked these: buffers are now proxied like most other objects.
+In simple use cases they can be converted with a copy using {any}`PyProxy.toJs`
+and {any}`JsProxy.to_py`. We added new APIs {any}`PyProxy.getBuffer`,
+{any}`JsProxy.assign`, and {any}`JsProxy.assign_to` which give more fine-grained
+control, though they are not yet as ergonomic as they could be.
+
+(Implemented in PRs {pr}`1215`, {pr}`1376`, and {pr}`1411`)
+
+## Maintenance and Bug fixes
+Pyodide version 0.17.0 comes with an enormous amount of maintenance work. There
+is still a lot of work to do be done before Pyodide will be ready for a version
+1.0, but we made very significant headway. Significant improvements were made to
+every component of Pyodide, including the build system, the test suite and
+continuous integration, and the core Pyodide code.
+
+### Upstream Emscripten
+
+We finally completed the migration to the latest version of Emscripten
+(https://emscripten.org/) compiler toolchain which uses the upstream LLVM
+backend. This allows us to take advantage of  recent improvements to the
+toolchain reducing package size and execution time.
+
+For instance, the scipy package  shrank dramatically from 92 MB to 15 MB. Scipy
+is now automatically cached in browsers, greatly improving the usability of
+scientific Python packages that depend on scipy, such as scikit-image and
+scikit-learn. The size of the base Pyodide environment with only the CPython
+standard library shrank from 8.1 MB to 6.4 MB.
```

Comment:
Well to be precise this is more due to not vendoring jedi than to Emscripten update, but I wasn't sure how and where to phrase in the current narrative.

Reply:
> more due to not vendoring jedi

That's a bit disappointing.

Reply:
Yeah, I mean it does reduce the size of the generated wasm but in base pyodide by <0.5MB or so, but fair amount of size is due to `pyodide.asm.data` anyway.